### ISSUE TITLE
`used-before-assignment` with local variable annotations

### ISSUE THREAD

**Issue by cdce8p** [PyCQA/pylint#7773]
```py
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from t import SomeAlias

def f():
    x: SomeAlias = 2  # used-before-assignment
    print(tag)
```
```
Using variable 'SomeAlias' before assignment (used-before-assignment)
```

Annotations on `local` variables are never evaluated. Thus `used-before-assignment` shouldn't be emitted.
We already check if an annotation is stringified, but that isn't even important here.

However, these should still be errors
```py
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from t import OtherAlias, OtherAlias2

y: OtherAlias = 2  # used-before-assignment -> correct!

class A:
    z: OtherAlias2 = 2  # used-before-assignment -> correct!
```

https://peps.python.org/pep-0526/#runtime-effects-of-type-annotations
https://stackoverflow.com/questions/41291892/variable-type-annotation-nameerror-inconsistency

**Comment by jacobtylerwalls** [comment]
Duplicate of #7609
